content/posts/irods-internship-summer-2024.md

```diff
@@ -0,0 +1,77 @@
+Title: iRODS Internship: Summer 2024
+Date: 2024-01-05 08:00
+Author: Terrell Russell
+Slug: irods-internship-summer-2024
+Status: published
+
+## Join Us
+
+The iRODS Consortium is looking for qualified, motivated candidates for a paid, full-time, remote, summer internship (probably three months, June-August).  We have brainstormed five separate projects, each appropriate for a summer of focus.  Each project is real work that we would like to have done and will be an important part of the iRODS ecosystem in the near future.
+
+This internship would provide experience with working in a production-quality software engineering environment with global reach, mentorship and periodic code reviews, direct feedback from community members, and a final presentation about the work and its impact.
+
+This internship is open to United States citizens who are students at least 18 years of age.
+
+The Integrated Rule-Oriented Data System (iRODS) is open source data management software used by research, commercial, and governmental organizations worldwide.
+
+The iRODS Consortium brings together businesses, research organizations, universities, and government agencies to ensure the sustainability of iRODS by guiding further development of the software, growing the user and developer communities, and facilitating iRODS support, education, and collaboration opportunities.  The iRODS Consortium fields a team of software developers, application engineers, and support staff housed at RENCI at the University of North Carolina at Chapel Hill.
+
+[https://irods.org](https://irods.org)
+
+[https://irods.org/documentation](https://irods.org/documentation)
+
+[https://irods.org/ugm2023](https://irods.org/ugm2023)
+
+[https://github.com/irods](https://github.com/irods)
+
+## Five Projects
+
+In no particular order:
+
+
+### 1) Create an iRODS database plugin template repository (C++)
+
+The iRODS Development Team has built project templates as examples for four of the seven plugin interfaces into iRODS and one more as an example C++ client.  The next plugin we'd like to have a template for is the database plugin.  iRODS currently supports PostgreSQL, MySQL/MariaDB, and Oracle, but the code has not been pulled out into separate repositories, yet.  This example repository would pave the way for increased modularity of the codebase.
```

Comment:
An even bigger benefit would be the ability for devs (us included) to experiment adding support for other databases and systems to iRODS. I feel that is the primary focus of the template project effort. _I'm looking at the last statement specifically here._

I don't know that we'd ever separate the three backends we currently support, but I can see us rewriting the implementation using nanodbc and a couple if-stmts to handle database specific behavior.

Reply:
ah, very good.  will reword a bit.

Reply:
reworded in second commit.

Reply:
Looks good.

Reply:
good... squashing.